pkg/scheduler/generic_scheduler_test.go

```diff
@@ -1335,15 +1347,21 @@ func TestZeroRequest(t *testing.T) {
 				t.Fatalf("error creating framework: %+v", err)
 			}
 
-			scheduler := NewGenericScheduler(
+			scheduler := newScheduler(
+				nil,
+				nil,
```

Comment:
It's hard to maintain this function. Suggest to introduce `Builder` pattern https://github.com/kubernetes/kubernetes/blob/a83bb7746c18caa53fafdeaa882cbb21a120547c/staging/src/k8s.io/kubectl/pkg/cmd/get/get.go#L639-L650

or closures like
https://github.com/kubernetes/kubernetes/blob/a83bb7746c18caa53fafdeaa882cbb21a120547c/pkg/scheduler/scheduler.go#L117-L125
For not required params, lets mark it as an `option`.

Glad to hear your advices @ahg-g

Reply:
we used the with pattern in other places, let us stick with that, but I would do it in a followup PR.

Reply:
But lets hold off on that, what I think we need to do first as followup is to get rid of the "configurator" completely, and perhaps unroll newScheduler with scheduler.New (which already applies the with pattern)

Reply:
sounds good.